src/wwwroot/app/views/vacancy/vacancy.edit.controller.js

```diff
@@ -89,6 +93,23 @@ export default function VacancyController(
       $state.go('vacancyEdit', {_data: null, vacancyId: null});
    }
 
+   function goToChildVacancy(vacancy) {
+      $state.go('vacancyView', {_data: null, vacancyId: vacancy.id});
+   }
+
+   function removeChildVacancy(vacancy) {
+      UserDialogService.confirm($translate.instant('VACANCY.VACANCY_REMOVE_MESSAGE')).then(() => {
+         VacancyService.remove(vacancy).then((responseVacancy) => {
```

Comment:
again and again
change the scope of promise resolution, return the result and use it further